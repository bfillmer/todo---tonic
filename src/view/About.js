
import React from 'react'
import Link from 'redux-first-router-link'

import {routeHome} from 'actions'

import {Section} from 'view/Shared/Section'
import {H1, P} from 'view/Shared/Typography'

export const About = () => (
  <Section>
    <H1>About</H1>
    <P>Head to the <Link to={routeHome()}>Home Page</Link>.</P>
  </Section>
)
