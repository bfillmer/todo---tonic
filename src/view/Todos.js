
import React from 'react'

import {Section} from 'view/Shared/Section'
import {H1} from 'view/Shared/Typography'

export const Todos = () => (
  <Section>
    <H1>Todos</H1>
  </Section>
)
