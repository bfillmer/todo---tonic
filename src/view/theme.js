
const richBlack = `rgba(4, 15, 15, 1)`
const platinum = `rgba(226, 239, 222, 1)`
const greenMunsell = `rgba(13, 171, 118, 1)`
const internationalOrange = `rgba(193, 41, 46, 1)`

export const theme = {
  colors: {
    primary: platinum,
    secondary: greenMunsell,
    black: richBlack,
    warning: internationalOrange
  },
  bottomMargin: '1.5rem'
}

// Simple helper function, takes in any number of props mapping to properties within the theme
// object and returns the value.
export const getTheme = (...props) => ({theme}) => props.reduce((t, p) => t[p], theme)
