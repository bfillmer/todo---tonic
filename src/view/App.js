
import React from 'react'
import styled, {ThemeProvider} from 'styled-components'

import {Header} from 'view/Shared/Header'
import {Routes} from 'view/Routes'

import {theme} from 'view/theme'

// GLOBAL WRAPPER
const Wrapper = styled.div`
  height: 100vh;
`

export const App = () => (
  <ThemeProvider theme={theme}>
    <Wrapper>
      <Header />
      <Routes />
    </Wrapper>
  </ThemeProvider>
)
