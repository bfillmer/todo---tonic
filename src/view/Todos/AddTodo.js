
import React, {Component} from 'react'
import styled from 'styled-components'

import {getTheme} from 'view/theme'

const Input = styled.input`
  width: 100%;
  font-size: 1.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  border: 1px solid ${getTheme('colors', 'black')};
  border-radius: 0.25rem;
  &::placeholder {
    color: #ccc;
  }
`

export class AddTodo extends Component {
  constructor () {
    super()
    this.state = {
      text: ''
    }
    this.handleText = this.handleText.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleText (e) {
    this.setState({text: e.target.value})
  }

  handleSubmit (e) {
    e.preventDefault()
    this.props.addTodo(this.state.text)
    this.setState(state => ({text: ''}))
  }

  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <Input
          type='text'
          value={this.state.text}
          onChange={this.handleText}
          placeholder='I need to...'
        />
      </form>
    )
  }
}
