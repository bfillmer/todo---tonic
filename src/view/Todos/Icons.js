
import React from 'react'
import styled from 'styled-components'

import {getTheme} from 'view/theme'

const Icon = styled.svg`
width: 1rem;
height: 1rem;
`

// Check Mark
const InProgressSvg = styled(Icon)`
  fill: ${getTheme('colors', 'secondary')};
`

export const InProgress = () => (
  <InProgressSvg viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'>
    <path d='M0 0h24v24H0z' fill='none' />
    <path d='M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z' />
  </InProgressSvg>
)

// Completed
const CompletedSvg = styled(Icon)`
  fill: ${getTheme('colors', 'secondary')};
`

export const Completed = () => (
  <CompletedSvg viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'>
    <path d='M0 0h24v24H0z' fill='none' />
    <path d='M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z' />
  </CompletedSvg>
)

// Trash Can
const RemoveSvg = styled(Icon)`
  fill: ${getTheme('colors', 'warning')};
`

export const Remove = () => (
  <RemoveSvg viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'>
    <path d='M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z' />
    <path d='M0 0h24v24H0z' fill='none' />
  </RemoveSvg>
)
