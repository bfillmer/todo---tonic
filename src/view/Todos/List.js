
import React from 'react'
import styled from 'styled-components'

import {COMPLETED} from 'types'

import {Remove} from 'view/Todos/Icons'
import {getTheme} from 'view/theme'

export const List = styled.ul`
  padding-top: 2rem;
  padding-bottom: 2rem;
`

// Children styles are applied here as they are specific to this implementation of the list item.
const Li = styled.li`
  display: flex;
  align-items: center;
  justify-items: space-between;
  font-size: 1rem;
  border-bottom: 1px solid ${getTheme('colors', 'primary')};
  &:hover {
    background-color: ${getTheme('colors', 'primary')};
  }
  &:first-child {
    border-top: 1px solid ${getTheme('colors', 'primary')};
  }
`

const Button = styled.button`
  cursor: pointer;
  outline: 0;
  font-size: 1rem;
  text-align: left;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
  border: none;
  background: none;
`

const Complete = styled(Button)`
  flex: 1;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  text-decoration: ${({status}) => status === COMPLETED ? 'line-through' : 'none'};
`

export const Todo = ({id, text, status, completeTodo, removeTodo}) => (
  <Li>
    <Complete status={status} onClick={() => completeTodo(id)}>{text}</Complete>
    <Button onClick={() => removeTodo(id)}><Remove /></Button>
  </Li>
)
