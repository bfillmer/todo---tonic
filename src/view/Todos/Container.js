
import React from 'react'
import {connect} from 'react-redux'

import {getAllTodos} from 'selectors'
import {addTodo, completeTodo, removeTodo} from 'actions'

import {Section} from 'view/Shared/Layout'
import {AddTodo} from 'view/Todos/AddTodo'
import {List, Todo} from 'view/Todos/List'

const mapStateToProps = state => ({
  todos: getAllTodos(state)
})

const mapDispatchToProps = {
  addTodo,
  completeTodo,
  removeTodo
}

export const Container = ({addTodo, completeTodo, removeTodo, todos}) => {
  const todoActions = {removeTodo, completeTodo}
  const renderTodos = todos.map(t => <Todo key={t.id} {...t} {...todoActions} />)
  return (
    <Section>
      <AddTodo addTodo={addTodo} />
      <List>
        {renderTodos}
      </List>
    </Section>
  )
}

export const Todos = connect(mapStateToProps, mapDispatchToProps)(Container)
