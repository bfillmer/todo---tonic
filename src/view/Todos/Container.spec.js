/* global expect, test */

import React from 'react'
import renderer from 'react-test-renderer'
import {ThemeProvider} from 'styled-components'

import {Container} from 'view/Todos/Container'
import {theme} from 'view/theme'

// MOCK PROPS
const addTodo = () => null
const completeTodo = () => null
const removeTodo = () => null
const todos = [
  {
    'id': 1,
    'status': 'IN_PROGRESS',
    'text': 'A simple todo item.'
  }
]

test('[SNAP] Todos Container', () => {
  const props = {addTodo, completeTodo, removeTodo, todos}
  const expected = renderer.create(
    <ThemeProvider theme={theme}>
      <Container {...props} />
    </ThemeProvider>
  ).toJSON()
  expect(expected).toMatchSnapshot()
})
