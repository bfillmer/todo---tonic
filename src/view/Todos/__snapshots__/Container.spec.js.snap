// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAP] Todos Container 1`] = `
<section
  className="sc-bdVaJa hGVrDW"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="sc-bwzfXH juMeWl"
      onChange={[Function]}
      placeholder="I need to..."
      type="text"
      value=""
    />
  </form>
  <ul
    className="sc-ifAKCX cgkBfR"
  >
    <li
      className="sc-EHOje bRpoAK"
    >
      <button
        className="sc-gzVnrw fpSixZ sc-bZQynM kDYktl"
        onClick={[Function]}
      >
        A simple todo item.
      </button>
      <button
        className="sc-bZQynM kDYktl"
        onClick={[Function]}
      >
        <svg
          className="sc-bxivhb iWLryQ sc-htpNat ldqdhg"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </button>
    </li>
  </ul>
</section>
`;
